pkg/helm/controller.go

```diff
@@ -150,21 +164,46 @@ func (c *Controller) OnHelmRemove(key string, chart *helmv1.HelmChart) (*helmv1.
 	return newChart, c.apply.WithOwner(newChart).Apply(objectset.NewObjectSet())
 }
 
+func (c *Controller) OnConfChange(key string, conf *helmv1.HelmChartConfig) (*helmv1.HelmChartConfig, error) {
+	if chart, err := c.helmController.Cache().Get(conf.Namespace, conf.Name); err != nil {
+		if !errors.IsNotFound(err) {
+			return conf, err
```

Comment:
not sure if we should send back an error here, considering a scenario where the user adds the config before adding the chart, the controller would keep erroring since it cant find the chart, should we just return nil and just wait for the chart to be created and the controller will handle that then

Reply:
this return is inside `if !errors.IsNotFound(err)` so it won't propagate the failure up if the chart is missing.